app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFragment.kt

```diff
@@ -61,16 +76,59 @@ class TabsTrayFragment : AppCompatDialogFragment(), TabsTrayInteractor {
         val containerView = inflater.inflate(R.layout.fragment_tab_tray_dialog, container, false)
         val view: View = LayoutInflater.from(containerView.context)
             .inflate(R.layout.component_tabstray2, containerView as ViewGroup, true)
+        val activity = activity as HomeActivity
 
         behavior = BottomSheetBehavior.from(view.tab_wrapper)
 
+        navigationInteractor =
+            DefaultNavigationInteractor(
+                browserStore = activity.components.core.store,
+                navController = findNavController(),
+                metrics = activity.components.analytics.metrics,
+                dismissTabTray = ::dismissAllowingStateLoss,
+                dismissTabTrayAndNavigateHome = ::dismissTabTrayAndNavigateHome
+            )
+
         return containerView
     }
 
     @ExperimentalCoroutinesApi
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
+        view.tab_tray_overflow.setOnClickListener {
```

Comment:
nit: can we move the construction and setup of the menu to separate class/method? The fragment's `onViewCreated` method will get big very quickly if we add everything in there.

Reply:
Agreed.